### PR TITLE
hal/ia32: cast operands to correct sizes

### PR DESCRIPTION
**Pull request by badochov**
Copy "in" related function from efi.
Newer binutils versions check closely if asm instructions get operands of correct size.

JIRA: RTOS-912

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (refactoring, style fixes, git/CI config, submodule management, no code logic changes)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by badochov** [comment]
```c
	__asm__ volatile(
		"inw %1, %0; "
	: "=a" (val)
	: "d" ((uint16_t)addr));
```
generates
`inw dx, ax; `

whereas
```c
	__asm__ volatile(
		"inw %1, %0; "
	: "=a" (val)
	: "d" (addr));
```
generares
`inw edx,ax;`

**Comment by badochov** [comment]
BTW, why is this code copied 3 times in plo?

File: hal/ia32/cpu.h [hunk 1]
```diff
@@ -36,66 +39,84 @@ static inline void hal_cpuInstrBarrier(void)
 }
 
 
-static inline unsigned char hal_inb(void *addr)
+static inline u8 hal_inb(void *addr)
 {
-	unsigned char val;
+	u32 ioport = (u32)addr;
```
Comment (agkaminski) [inline hal/ia32/cpu.h:44]: IMHO double cast here is somewhat better (like you did in the kernel). Perhaps move API `void *` -> `u16`, if easily done

**Review by agkaminski — APPROVED** [review]
Thanks!

**Pull request merged**